libs/platform/ebpf_hash_table.c

```diff
@@ -3,52 +3,175 @@
 
 #include "ebpf_platform.h"
 
+typedef struct _ebpf_hash_bucket_header
+{
+    struct _ebpf_hash_bucket_header* next;
+    uint8_t data[1];
+} ebpf_hash_bucket_header_t;
+
 struct _ebpf_hash_table
 {
-    struct _RTL_AVL_TABLE avl_table;
-    ebpf_hash_table_compare_result_t (*compare_function)(const uint8_t* key1, const uint8_t* key2);
+    uint32_t bucket_count;
+    uint32_t entry_count;
+    uint32_t seed;
     size_t key_size;
     size_t value_size;
     void* (*allocate)(size_t size);
     void (*free)(void* memory);
+    void (*extract)(const uint8_t* value, const uint8_t** data, size_t* num);
```

Comment:
prefer separate compare_cb and optional hash_cb. If hash_cb == NULL, use murmur.

Reply:
Add it when and if the need arises.